core/src/main/java/org/apache/accumulo/core/conf/ClientConfigGenerate.java

```diff
@@ -0,0 +1,193 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.conf;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.FileNotFoundException;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+
+import com.google.common.collect.Sets;
+
+/**
+ * Generates client-properties.md for documentation on Accumulo website and accumulo-client.properties for Accumulo distribution tarball
+ */
+class ClientConfigGenerate {
+
+  private abstract class Format {
+
+    abstract void beginSection(String section);
+
+    abstract void pageHeader();
+
+    abstract void property(ClientProperty prop);
+
+    void generate() {
+      pageHeader();
+
+      generateSection("Instance", "instance.");
+      generateSection("Authentication", "auth.", "auth.method", "auth.username");
+      generateSection("Batch Writer", "batch.writer.");
+      generateSection("SSL", "ssl.");
+      generateSection("SASL", "sasl.");
+      generateSection("Tracing", "trace.");
+
+      doc.close();
+    }
+
+    void generateSection(String section, String prefix, String... prefixProps) {
+      beginSection(section);
+      for (String prop : prefixProps) {
+        ClientProperty cp = sortedProps.get(prop);
+        if (cp != null) {
+          property(cp);
+        }
+      }
+      Set<String> prefixSet = Sets.newHashSet(prefixProps);
+      for (ClientProperty prop : sortedProps.values()) {
+        if (prop.getKey().startsWith(prefix) && !prefixSet.contains(prop.getKey())) {
+          property(prop);
+        }
+      }
+    }
+
+    void generateSection(String section, String prefix) {
+      generateSection(section, prefix, "");
+    }
+  }
+
+  private class Markdown extends Format {
+
+    @Override
+    void beginSection(String section) {}
+
+    @Override
+    void pageHeader() {
+      doc.println("---");
+      doc.println("title: Client Properties");
+      doc.println("category: development");
+      doc.println("order: 9");
+      doc.println("---\n");
+      doc.println("<!-- WARNING: Do not edit this file. It is a generated file that is copied from Accumulo build (from core/target/generated-docs) -->\n");
+      doc.println("Below are properties set in `accumulo-client.properties` that configure Accumulo clients:\n");
+      doc.println("| Property | Default value | Description |");
```

Comment:
A since column would be nice.  Maybe not now, but in the future.

Reply:
Added in ce512a93bfa9